nestkernel/per_thread_bool_indicator.h

```diff
@@ -133,6 +163,7 @@ class PerThreadBoolIndicator
 
 private:
   std::vector< BoolIndicatorUInt64 > per_thread_status_;
+  std::atomic< int > size_ { 0 }, are_true_ { 0 };
```

Comment:
We generally avoid multiple definitions in a single statement. 

```suggestion
  std::atomic< int > size_ { 0 };
  
  /** Number of per-thread indicators currently true
   *
   * are_true_ == 0 -> all are false
   * are_true_ == size_  -> all are true
   * 0 < are_true_ < size_  -> some true, some false
   */
  std::atomic< int > are_true_ { 0 };
```

Another question: Does `size_` need to be atomic, or is there any advantage to it being atomic, given that it is only set by `initialize()` and then just read later?

Reply:
I think, size_ doesn't need to be atomic